pandas/_libs/tslib.pyx

```diff
@@ -2230,6 +2230,11 @@ cpdef array_with_unit_to_datetime(ndarray values, unit, errors='coerce'):
     m = cast_from_unit(None, unit)
 
     if is_raise:
+        from pandas.core.dtypes.common import is_bool_dtype
+
+        if is_bool_dtype(values):
```

Comment:
If I raise only on non-numeric, boolean values go through and do not raise error